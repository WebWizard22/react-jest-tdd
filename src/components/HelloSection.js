import React from 'react';
import { connect } from 'react-redux';

export class HelloSection extends React.Component {
  render() {
    return (
      <section>
        <h1>Hello!</h1>
        <h2>Username: {this.props.username}</h2>
        <p>I have some todos below</p>
        <ul>
          { this.props.todos && this.props.todos.map(todo => (
            <li key={todo.id}>
              <span>{todo.task}</span>
              <button
                className="delete-btn"
                onClick={() => this.removeTodo(todo.id)}
              >
                DELETE
              </button>
            </li>
          )) }
        </ul>
      </section>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    todos: state.todos,
  };
};

export default connect(mapStateToProps, null)(HelloSection);
