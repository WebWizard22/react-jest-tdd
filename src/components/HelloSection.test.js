import React from 'react';
import { shallow, mount } from 'enzyme';

import HelloSection from './HelloSection';

describe('<HelloSection> component ', () => {

  /*
  ============
  OUTPUT TESTS
  ============
  */
  const helloSectionWrapper = shallow(<HelloSection />);
  it('renders correctly', () => {
    expect(helloSectionWrapper).toHaveLength(1);
  });

  it('renders a <h1>', () => {
    const helloSectionH1 = helloSectionWrapper.find('h1');
    expect(helloSectionH1).toHaveLength(1);
  });

  it('renders a paragraph <p>', () => {
    const helloSectionParagraph = helloSectionWrapper.find('p');
    expect(helloSectionParagraph).toHaveLength(1);
  });

  it('renders an <ul>', () => {
    const helloSectionUl = helloSectionWrapper.find('ul');
    expect(helloSectionUl).toHaveLength(1);
  });

  it('renders three <li>s inside <ul>', () => {
    const helloSectionUl = helloSectionWrapper.find('ul');
    const LiInsidehelloSectionUl = helloSectionUl.find('li');
    expect(LiInsidehelloSectionUl).toHaveLength(3);
  });

  /*
  ============
  STATE TESTS
  ============
  */
  it('have todos state which consists of three items', () => {
    const todosState = helloSectionWrapper.state('todos');
    expect(todosState).toHaveLength(3);
  });

  /*
  ============
  EVENTS TESTS
  ============
  */
  it('adds a new todo to the state when addTodo function called', () => {
    const newTodo = {
      id: 4,
      task: 'TDD!',
      completed: false,
    };

    helloSectionWrapper.instance().addTodo(newTodo);
    const todosState = helloSectionWrapper.state('todos');
    expect(todosState).toHaveLength(4);
  });

  it('deletes a todo from a state when removeTodo function called', () => {
    const todoToDeleteId = 1;
    helloSectionWrapper.instance().removeTodo(todoToDeleteId);
    const todosState = helloSectionWrapper.state('todos');
    expect(todosState).toHaveLength(3);
  });

  it('deletes a todo when DELETE button clicked', () => {
    const buttonToClick = helloSectionWrapper.find('button').at(0);
    // let's click the button!
    buttonToClick.simulate('click');
    const todosState = helloSectionWrapper.state('todos');
    expect(todosState).toHaveLength(2);
  });
});
