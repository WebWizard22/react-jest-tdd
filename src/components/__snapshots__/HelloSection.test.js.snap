// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectedHelloSection> redux-connected component   renders correctly and matched snapshot 1`] = `
<section>
  <h1>
    Hello!
  </h1>
  <h2>
    Username: 
  </h2>
  <p>
    I have some todos below
  </p>
  <ul>
    <li>
      <span>
        My first todo
      </span>
      <button
        className="delete-btn"
        onClick={[Function]}
      >
        DELETE
      </button>
    </li>
  </ul>
</section>
`;

exports[`<HelloSection> component  renders correctly and matched snapshot 1`] = `
<section>
  <h1>
    Hello!
  </h1>
  <h2>
    Username: 
  </h2>
  <p>
    I have some todos below
  </p>
  <ul />
</section>
`;
